pkg/hintrunner/zero/zerohint_blake.go

```diff
@@ -215,3 +216,143 @@ func createBlake2sFinalizeHinter(resolver hintReferenceResolver) (hinter.Hinter,
 
 	return newBlake2sFinalizeHint(blake2sPtrEnd, nPackedInstances, inputBlockFelt), nil
 }
+
+// Blake2sCompute hint computes the blake2s compress function and fills the value in the right position.
+//
+// `newBlake2sComputeHint` takes 1 operander as an argument
+//   - `output` should point to the middle of an instance, right after initial_state, message, t, f,
+//     which should all have a value at this point, and right before the output portion which will be
+//     written by this function.
+func newBlake2sComputeHint(output hinter.ResOperander) hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "Blake2sCompute",
+		Op: func(vm *VM.VirtualMachine, _ *hinter.HintRunnerContext) error {
+			//> from starkware.cairo.common.cairo_blake2s.blake2s_utils import compute_blake2s_func
+			//> compute_blake2s_func(segments=segments, output_ptr=ids.output)
+
+			// Expanding the compute_blake2s_func function here:
+			//> def compute_blake2s_func(segments: MemorySegmentManager, output_ptr: RelocatableValue):
+			//>    h = segments.memory.get_range(output_ptr - 26, 8)
+			//>    message = segments.memory.get_range(output_ptr - 18, 16)
+			//>    t = segments.memory[output_ptr - 2]
+			//>    f = segments.memory[output_ptr - 1]
+			//>    new_state = blake2s_compress(
+			//>        message=message,
+			//>        h=h,
+			//>        t0=t,
+			//>        t1=0,
+			//>        f0=f,
+			//>        f1=0,
+			//>    )
+			//>    segments.write_arg(output_ptr, new_state)
+
+			output, err := hinter.ResolveAsAddress(vm, output)
+			if err != nil {
+				return err
+			}
+
+			//> h = segments.memory.get_range(output_ptr - 26, 8)
+			hSegmentInput := new(mem.MemoryAddress)
+			err = hSegmentInput.Sub(output, new(fp.Element).SetUint64(26))
+			if err != nil {
+				return err
+			}
+			h, err := vm.Memory.GetConsecutiveMemoryValues(*hSegmentInput, 8)
+			if err != nil {
+				return err
+			}
+			var hUint32 [8]uint32
+			for i := 0; i < 8; i++ {
+				value, err := hintrunnerUtils.ToSafeUint32(&h[i])
+				if err != nil {
+					return err
+				}
+				hUint32[i] = value
+			}
+
+			//> message = segments.memory.get_range(output_ptr - 18, 16)
+			messageSegmentInput := new(mem.MemoryAddress)
+			err = messageSegmentInput.Sub(output, new(fp.Element).SetUint64(18))
+			if err != nil {
+				return err
+			}
+			message, err := vm.Memory.GetConsecutiveMemoryValues(*messageSegmentInput, 16)
+			if err != nil {
+				return err
+			}
+			var messageUint32 []uint32
+			for i := 0; i < 16; i++ {
+				value, err := hintrunnerUtils.ToSafeUint32(&message[i])
+				if err != nil {
+					return err
+				}
+				messageUint32 = append(messageUint32, value)
```

Comment:
since the size of the array is 16, why append instead of setting to an index just like in hUint32

Reply:
Ah because the method signature of the `Blake2sCompress` function expects that